File: windows/src/ConnectorControl.Core/State/EnvRow.cs
namespace ConnectorControl.Core.State;

/// <summary>
/// One environment-variable row (catalog §3.6). Rows carry a stable identity while the name is
/// edited; values are masked unless <see cref="Revealed"/> (only freshly added rows start revealed).
/// </summary>
public sealed class EnvRow : ObservableObject
{
    private string name;
    private string value;
    private bool revealed;

    public EnvRow(string name, string value)
    {
        this.name = name;
        this.value = value;
    }

    public Guid Id { get; } = Guid.NewGuid();

    public string Name { get => name; set => Set(ref name, value); }

    public string Value { get => value; set => Set(ref this.value, value); }

    public bool Revealed { get => revealed; set => Set(ref revealed, value); }
}

File: Sources/ConnectorControlState/EnvRow.swift
import Foundation

/// One environment-variable row (catalog §3.6). Rows carry a stable identity
/// while the name is edited — a dictionary key cannot, since each keystroke
/// would re-key the dictionary, re-sort the list, and drop field focus.
/// Values are masked unless `revealed` (only freshly added rows start revealed).
public struct EnvRow: Identifiable, Equatable, Sendable {
    public let id: UUID
    public var name: String
    public var value: String
    public var revealed: Bool

    public init(name: String, value: String, revealed: Bool = false) {
        id = UUID()
        self.name = name
        self.value = value
        self.revealed = revealed
    }
}
